src/controller/python/chip/ChipDeviceCtrl.py

```diff
@@ -408,11 +408,11 @@ def ConnectBLE(self, discriminator, setupPinCode, nodeid):
         self._ChipStack.CallAsync(
             lambda: self._dmLib.pychip_DeviceController_ConnectBLE(
                 self.devCtrl, discriminator, setupPinCode, nodeid)
-        ).raise_on_error()
+        )
         if not self._ChipStack.commissioningCompleteEvent.isSet():
             # Error 50 is a timeout
-            return False
-        return self._ChipStack.commissioningEventRes.is_success
+            return False, PyChipError(0x32)
+        return self._ChipStack.commissioningEventRes.is_success, self._ChipStack.commissioningEventRes
```

Comment:
This change to return 2 things is very disruptive, and also a bit clunky. Suggest having a error object (`CommissioningStatus` or such) that has a truth value (`def __bool__(self)` special method) that returns based on the `self._ChipStack.commissioningEventRes.is_success` , and allows getting the underlying PyChipError as a property.

That way, none of the calling code has to get 2 values from a tuple, all the `if` still work, and nevertheless you can access the error code.